tests/rados/test_scrub_deepscrub_timecheck.py

```diff
@@ -0,0 +1,190 @@
+"""
+BZ#2330755: Verification of  deep scrub taking too long under mclock I/O scheduler .
+"""
+
+import datetime
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Bug Verfication steps:
+    1. Create a replicate pool with single PG
+    2. Push the data into the pool
+    3. Identify the active OSDs of the pool
+    4. Set the osd_mclock_max_capacity_iops_hdd with less values
+    5. Start deep-scrub and check the time.The wait time in the script is 60 minutes
+    6. Set the osd_mclock_force_run_benchmark_on_init to true
+    7. Remove the osd_mclock_max_capacity_iops_hdd value and restart the OSDs
+    8. Check the deep-scrub completed in less time
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    pool_obj = PoolFunctions(node=cephadm)
+    ceph_nodes = kw.get("ceph_nodes")
+    osd_list = []
+    mclock_max_capacity_values = ["2.985434", "0.198044", "0.198104"]
+
+    try:
+        pool_name = config["pool_name"]
+        method_should_succeed(rados_object.create_pool, **config)
+        rados_object.bench_write(pool_name=pool_name, rados_write_duration=60)
+        # Get the OSD list
+        for node in ceph_nodes:
+            if node.role == "osd":
+                node_osds = rados_object.collect_osd_daemon_ids(node)
+                osd_list = osd_list + node_osds
+            log.info(f"The OSDs in the cluster are-{osd_list}")
+
+            # Get the osd_mclock_max_capacity_iops_hdd values for all OSDs
+        for osd_id in osd_list:
+            capacity_iops_value = mon_obj.show_config(
+                daemon="osd", id=osd_id, param="osd_mclock_max_capacity_iops_hdd"
+            )
+            log.info(
+                f"The osd-{osd_id} osd_mclock_max_capacity_iops_hdd value - {capacity_iops_value}"
```

Comment:
If all the later operations are going to be executed on the 3 OSDs part of the acting set, why do we need to capture the `osd_mclock_max_capacity_iops_hdd` for all the OSDs?

Reply:
This is for logging purposes. If  it is required I  will change the code to show  only for the 3 OSDs not for all OSDs